examples/gltf_new/main.rs

```diff
@@ -0,0 +1,260 @@
+                        .with(scene_handle)
+                        .with(GlobalTransform::default())
+                        .build();
+
+                    true
+                }
+
+                Some(Completion::Failed) => {
+                    println!("Error: {:?}", self.progress.as_ref().unwrap().errors());
+                    return Trans::Quit;
+                }
+            };
+            if remove {
+                self.progress = None;
+            }
+            if self.entity.is_none() {
+                if let Some(entity) = data.world
+                    .exec(|finder: TagFinder<AnimationMarker>| finder.find())
+                {
+                    self.entity = Some(entity);
+                    self.initialised = true;
+                }
+            }
+        }
+        data.data.update(&data.world);
+        Trans::None
+    }
+}
+
+fn toggle_or_cycle_animation(
+    entity: Option<Entity>,
+    scene: &mut Scene,
+    sets: &ReadStorage<AnimationSet<usize, Transform>>,
+    controls: &mut WriteStorage<AnimationControlSet<usize, Transform>>,
+) {
+    if let Some((entity, Some(animations))) = entity.map(|entity| (entity, sets.get(entity))) {
+        if animations.animations.len() > scene.animation_index {
+            let animation = animations.animations.get(&scene.animation_index).unwrap();
+            let mut set = get_animation_set::<usize, Transform>(controls, entity).unwrap();
+            if set.has_animation(scene.animation_index) {
+                set.toggle(scene.animation_index);
+            } else {
+                println!("Running animation {}", scene.animation_index);
+                set.add_animation(
+                    scene.animation_index,
+                    animation,
+                    EndControl::Normal,
+                    1.0,
+                    AnimationCommand::Start,
+                );
+            }
+            scene.animation_index += 1;
+            if scene.animation_index == 6 {
+                scene.animation_index = 7;
+            }
+            if scene.animation_index >= animations.animations.len() {
+                scene.animation_index = 0;
+            }
+        }
+    }
+}
+
+fn main() -> Result<(), amethyst::Error> {
+    amethyst::start_logger(Default::default());
+
+    let path = format!(
+        "{}/examples/gltf/resources/display_config.ron",
+        env!("CARGO_MANIFEST_DIR")
+    );
+
+    let resources_directory = format!("{}/examples/assets/", env!("CARGO_MANIFEST_DIR"));
+
+    let game_data = GameDataBuilder::default()
+        .with(
+            PrefabLoaderSystem::<ScenePrefabData>::default(),
+            "scene_loader",
+            &[],
+        )
+        .with(
+            GltfSceneLoaderSystem::default(),
+            "gltf_loader",
+            &["scene_loader"],
+        )
+        .with_basic_renderer(
+            path,
+            DrawPbmSeparate::new()
+                .with_vertex_skinning()
```

Comment:
This is where you request skinning, which requires vertex attributes for joint ids and weights.

Reply:
Oh right. It did not occur to me to check out the example main.rs... *Facepalm*

Should the engine or the game code try to analyze the GLTF data before loading it in order to request `.with_vertex_skinning()`? I like your suggested solution:
> An interim solution would be to add empty joint id/weight buffers if there is no such information in the gltf file.

It makes sense to me that the game code should not need to know the internals of each GLTF file that might be loaded, before they are loaded. Makes sense?

Reply:
Yep, the reverse works fine (models with joint ids/weights in passes without skinning), so just defaulting to empty buffers for those should solve the most immediate problem.

Reply:
Should I include the empty buffers change in this PR too? In that case could you point me in the right direction? :)